org.metafacture.fix/src/main/java/org/metafacture/metafix/FixMethod.java

```diff
@@ -0,0 +1,432 @@
+        map.put(currentKey, nested);
+        return map;
+    }
+
+    @SuppressWarnings("unchecked")
+    private static Object insertNested(final InsertMode mode, final Map<String, Object> map, final String value, final String currentKey,
+            final String[] remainingKeys) {
+        if (!map.containsKey(currentKey)) {
+            map.put(currentKey, new LinkedHashMap<String, Object>());
+        }
+        final Object nested = map.get(currentKey);
+        final Object result;
+        if (nested instanceof Map) {
+            result = insert(mode, (Map<String, Object>) nested, remainingKeys, value);
+        }
+        else if (nested instanceof List) {
+            processList(mode, value, remainingKeys, nested);
+            result = map.get(currentKey);
+        }
+        else {
+            throw new IllegalStateException(NESTED + nested);
+        }
+        return result;
+    }
+
+    @SuppressWarnings("unchecked")
+    private static void processList(final InsertMode mode, final String value, final String[] remainingKeys,
+            final Object nested) {
+        final List<Object> nestedList = (List<Object>) nested;
+        final Map<String, Object> nestedMap;
+        switch (remainingKeys[0]) {
+            case APPEND:
+                nestedMap = new LinkedHashMap<>();
+                nestedList.add(nestedMap);
+                insert(mode, nestedMap, Arrays.copyOfRange(remainingKeys, 1, remainingKeys.length), value);
+                break;
+            case LAST:
+                final Object last = nestedList.get(nestedList.size() - 1);
+                if (last instanceof Map) {
+                    nestedMap = (Map<String, Object>) last;
+                    insert(mode, nestedMap, Arrays.copyOfRange(remainingKeys, 1, remainingKeys.length), value);
+                }
+                break;
+            default:
+                nestedMap = new LinkedHashMap<>();
+                nestedList.add(nestedMap);
+                insert(mode, nestedMap, remainingKeys, value);
+                break;
+        }
+    }
+
+    @SuppressWarnings("checkstyle:ReturnCount")
+    static Object find(final Map<String, Object> map, final String[] keys) {
+        final String currentKey = keys[0];
+        if (!map.containsKey(currentKey)) {
+            return null;
+        }
+        if (keys.length == 1) {
+            return map.get(currentKey);
+        }
+        final String[] remainingKeys = Arrays.copyOfRange(keys, 1, keys.length);
+        return findNested(map, currentKey, remainingKeys);
+    }
+
+    private static Object findNested(final Map<String, Object> map, final String currentKey,
+            final String[] remainingKeys) {
+        final Object nested = map.get(currentKey);
+        // TODO: array of maps, like in insertNested
+        if (nested instanceof List) {
+            return ((List<?>) nested).stream().map(o -> findNested(map, currentKey, remainingKeys))
```

Comment:
What is this code supposed to do? (It's not currently covered by any tests.) I read it as: For each element in this list, find this list (at `currentKey`) in the map - thus **recursively** multiplying the list by its size!?! Was it maybe intended to find `remainingKeys` in each element (`o`)? (The comment above seems to indicate as much.)

Reply:
Yeah, weird, it should be doing something with each element `o`. Since no test fails without it, I think we should remove it and re-implement with a test later, if we need it.

Reply:
OK, I'll just remove it during my refactoring (#72) in an extra commit.